test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -148,6 +148,17 @@ public Node(
         }
 
         public synchronized void start(Version version) {
+            if (process != null) {
+                // ES was previously started, so we don't need to reconfigure, just start
```

Comment:
This assumption isn't true though. In some tests we _do_ need to reconfigure because we can provide settings that can change. When we "reconfigure" we are re-evaluating the node settings at the time of node start to pick up any changes such as settings provided by a `MutableSettingsProvider` or anything else passed as a `Supplier`.

This seems like an unnecessary optimization. What are we trying to avoid here?

Reply:
We want to start with the exact same configuration, we don't want to have configuration generated as it is below. Otherwise eg we get a new node id in a newly generated elasticsearch.yml.

Reply:
What exactly triggers a new node id? Can we perhaps just make it so the default behavior doesn't do that. I don't think generally want to do that in stateful tests.